### PR TITLE
Feature/env variables

### PR DESCRIPTION
**Pull request by subhendukundu**
Adding ability to add env variables using a `.env` file.
Please review this PR.

### CODE REVIEW

File: configs/webpack.client.config.js [hunk 1]
```diff
@@ -219,6 +219,17 @@ module.exports = (env, argv) => {
         filename: "[name].[chunkhash].css",
       }),
       new BuildManifestPlugin({ buildId }),
+      new webpack.DefinePlugin({
+        ...Object.keys(process.env).reduce(
```
Comment (jplhomer) [inline configs/webpack.client.config.js:223]: Could we refactor this to `filter` by `startsWith` and then `reduce`?

File: docs/data-fetching.md [hunk 2]
```diff
@@ -68,7 +68,7 @@ export async function getEdgeProps() {
 }
 ```
 
-There might be times when you **never want the page to be cached**. That's possible, too — just return `{ revalidate: 0 }` from `getEdgeProps` to tell Flareact to fetch a fresh page every single request.
+There might be times when you **never want the page to be cached**. That's possible, too — just return `{ revalidate: 0 }` from `getEdgeProps` to tell Flareact to fetch a fresh page every single request.
```
Comment (jplhomer) [inline docs/data-fetching.md:71]: What is changing in this diff?

**Review by jplhomer — COMMENTED** [review]
This is a neat idea, and similar to what Next.js does. However, it currently lacks support for Worker context (variables are only encoded in the client bundle). 

Additionally, I'm concerned about the interoperability with Cloudflare's secrets https://developers.cloudflare.com/workers/cli-wrangler/commands#secret

E.g. an ideal `env` file integration might also sync secrets with Cloudflare. 

Not 100% sure of the answers to these questions yet, but this is a good start!

File: docs/data-fetching.md (hunk 2 above)
Comment (subhendukundu) [reply docs/data-fetching.md:71]: I knew you would ask 🙈  I don't know why it's taking the blank space that we already had differently.

File: configs/webpack.client.config.js (hunk 1 above)
Comment (subhendukundu) [reply configs/webpack.client.config.js:223]: Sure, I can do that. I have a question about this. How would that help? 🙈  Performance?

**Review by subhendukundu — COMMENTED** [review]
For sure. I do have to check the support for Cloudflare's secrets. I am fairly new to Cloudflare as well. So I would need time for that.
But yes this is a good start to support `.env` as right away I don't have to push my secrets to git and use them in the repo, then definitely then `.env.local` and so on.

Should I add the docs for this? or this PR will need time to be merged?

File: configs/webpack.client.config.js (hunk 1 above)
Comment (jplhomer) [reply configs/webpack.client.config.js:223]: Just cleaner code but not a blocker!

**Review by jplhomer — APPROVED** [review]

**Pull request merged**